### ISSUE TITLE
An option to turn off visual effects

### ISSUE THREAD

**Issue by jan-kleks** [KyleU/solitaire.gg#5]
I mean things like when cards "shake" if you drag them.

**Comment by KyleU** [comment]
Easy enough. Other than the card "lean" when you drag them, are there other effects you'd want to disable (the particle effects when you score a card, etc)? Should they be separate options, or one bug "disable effects" toggle?

**Comment by jan-kleks** [comment]
Yup, I meant the particle effects as well as all dealing/shuffling animations. I guess "disable effects" would be OK.

**Comment by jan-kleks** [comment]
I wonder whether disabling these effects could help with performance issues on my desktop computer. It's an old 2008 iMac with a Radeon HD 2600 Pro GPU, the latest Firefox release. Unfortunately Apple is known for poor OpenGL support, but I did not think that a solitaire would be too much for my mac. :P

**Comment by jan-kleks** [comment]
Or maybe you could have a few visual effects options? Well, whatever. :P I wanted to say that you might want to consider toning down this "lean" effect even if it's turned "on". It still should be perfectly noticeable.

**Comment by jan-kleks** [comment]
I really like this automatic card resizing, but some users may find it confusing. You can also add it as one of the visual settings which could be disabled on demand.

**Comment by jan-kleks** [comment]
BTW, maybe solitaire.gg could automatically detect performance issues and disable graphical effects accordingly?

**Comment by jan-kleks** [comment]
Two more things regarding visual effects:

I'm still not sure I like the scoring animation, maybe there should be less particles or it should be more subtle (the border between effects being cool vs tacky is very thin, so it's good to be careful)?

Also, I think I have read somewhere that using assembly or C in web apps for certain code parts can help with performance issues.

**Comment by jan-kleks** [comment]
I have noticed significant performance improvements with the current beta version. Solitaire.gg is finally fully playable (even in a very big window!) on my old 2008 iMac with a Radeon HD 2600 Pro GPU. Well, dealing and card leaning animations are still not perfectly smooth though...  

Keep up good work!

**Comment by KyleU** [comment]
Thanks, I've been working to make it better. The lower resolution assets are the right solution, I'm working on that now. Right now I'm loading textures 16x larger than most screens need.

**Comment by jan-kleks** [comment]
I suggest you test solitaire.gg especially with desktop versions of Safari, it seems to have issues with WebGL performance, and if you make your app work in this browser, it might as well work everywhere. :P

**Comment by KyleU** [comment]
Would you let me know if https://beta.solitaire.gg/play/insaneklondike works in your browser? It's almost guaranteed to be unplayable, but it makes for a decent benchmark.

**Comment by jan-kleks** [comment]
It's really slow in the desktop Safari (this is what I'm using now).

**Comment by KyleU** [comment]
Tested in the latest Safari on my 2016 MacBook, it was at least as performant as Chrome. You can bring up the debug panel with `?`, what does your framerate display as?

**Comment by jan-kleks** [comment]
You mean the "?" key? The debug panel does not show up.
